### PR TITLE
🌱 ClusterClass: add documentation fields for patches and variables

### LINKED ISSUE #5649: ClusterClass: Add documentation fields for variables and patches
**User Story**

As a user of a ClusterClass it would be great if a ClusterClass provides additional information for patches and variables.

**Detailed Description**

One way to implement this would be to add additional fields (like `description`) to our Patch and Variable structs of the ClusterClass.

**Anything else you would like to add:**

For variables this can be implemented by introducing additional optional fields on the `JSONSchemaProps` struct. CRDs already have a few of them we just didn't add them to our copy of the `JSONSchemaProps` struct for the first iteration.

Some candidates are: (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go)
* Description
* Example
* ExternalDocs

It would be great to do it in a way which is compatible with CRDs and thus OpenAPI schema v3. This way those informations can be used later on by tools like CLIs or UIs to provide helpful information for users which want to create new Clusters or want to understand the variables of a ClusterClass (cf. what Swagger UI is doing with fields: https://petstore.swagger.io/#/pet/uploadFile)

/kind feature
/area topology

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Adds documentation fields which will make it easier for ClusterClass authors to document ClusterClasses.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5649

### CODE REVIEW

File: api/v1beta1/clusterclass_types.go [hunk 1]
```diff
@@ -261,6 +267,9 @@ type ClusterClassPatch struct {
 	// Name of the patch.
 	Name string `json:"name"`
 
+	// Description is a human-readable description of this patch.
+	Description string `json:"description,omitempty"`
```
Comment (sbueringer) [inline api/v1beta1/clusterclass_types.go:271]: I've only added it top-level for now. I'm not sure if we really need it on PatchDefinition and JSONPatch

**Comment by sbueringer** [comment]
/retest
flake

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**